### DISCUSSION TITLE
Can RandomTest mode use Tag-team mode?

### DISCUSSION THREAD

**Discussion by VinVulpis** [ikemen-engine/Ikemen-GO#1696] [Feature Request]
### Is your feature request related to a problem? Please describe.

I'm looking to get RandomTest mode to use 2v2 tag teams. The characters I'm making are meant to be used in tag-teams and I'm building my stuff around 2-character teams. 

I like the idea of randomized tag-teams being picked in an endless set of matches because I've been displaying my build at some conventions where I get tables to sell art and advertise my lil project. It's a good draw. I have more than 4 characters in the works now. I've been manually changing up the teams with 9999 rounds every so often while juggling other tasks. So if I could have the engine randomizing teams every match, endlessly, that'd be great for me. I could just set the display and not touch it.

I've poked around the code and I don't know a whole lot about Lua, just basics, all I've found was how to change the number of characters on each team, not the team-mode. 

It would also be handy if Demo mode could be tweaked for tag teams as well for when I eventually make other animated stuff for attract mode.

### Describe the solution you'd like

Solutions for me would be either to be told how to tweak the code myself, or if options for teammode could be added to it.

Or just simply a randomized never-ending watch mode.

### Search terms

randomtest demo

**Comment by lokpro** [comment]
it is there,

https://github.com/ikemen-engine/Ikemen-GO/blob/6831166711e438bd3720a06eb9bd104277b9d8b1/external/script/main.lua#L3703C1-L3708C5

this is my edit: (a randomized never-ending watch mode. set wait idle time to 0)
```lua
function main.f_demoStart()
	main.f_default()
	if motif.demo_mode.debuginfo == 0 and config.DebugKeys then
		setAllowDebugKeys(false)
		setAllowDebugMode(false)
	end
	main.lifebar.bars = motif.demo_mode.fight_bars_display == 1
	setGameMode('demo')
	
	local nPlayers = {math.random(1, 4), math.random(1, 4)}
	for iTeam = 1, 2 do
		setTeamMode(iTeam, math.random(0, 1) * 2+1, nPlayers[iTeam])
		for iPlayer = 1, nPlayers[iTeam] do
			-- setCom( iPlayer*2 + (iTeam-2), 8)
			local ch = main.t_randomChars[math.random(1, #main.t_randomChars)]
			selectChar(iTeam, ch, getCharRandomPalette(ch))
		end
	end
	for iPlayer = 1, 8 do
		setCom( iPlayer, 8)
	end

	local stage = start.f_setStage()
```

**Reply by VinVulpis** [reply]
Edited it to have 2v2 tagteams. Can just set the demo wait and fade times to 0 for events.
Thanks a ton!

**Comment by lokpro** [comment]
it was my request too~ https://github.com/ikemen-engine/Ikemen-GO/discussions/866

**Reply by VinVulpis** [reply]
I feel like doing it with randomtest would be cleaner looking without the refresh for 1 or 2 frames of the main menu, and needing to tweak engine settings, but that's just a small gripe, this works well enough.

I looked at randomtest.lua, but the terms seem a lil different there, so unsure what to change. I only got it to change team sizes, not the team mode though.
